### PR TITLE
Misc Test Improvements

### PR DESCRIPTION
**Pull request by ShikharJ**
Please review and suggest improvements.

### CODE REVIEW

**Comment by isuruf** [comment]
Thanks for the PR. Can you fix the merge conflicts?

**Comment by ShikharJ** [comment]
Updated @isuruf.

File: symengine/tests/basic/test_basic.cpp [hunk 1]
```diff
@@ -557,136 +559,138 @@ TEST_CASE("compare: Basic", "[basic]")
...
-    CHECK(x->compare(*z) == -1);
-    CHECK(y->compare(*x) == 1);
-    CHECK(y->compare(*z) == -1);
-    CHECK(z->compare(*x) == 1);
-    CHECK(z->compare(*y) == 1);
-
-    CHECK(i2->compare(*i2) == 0);
-    CHECK(i2->compare(*i3) == -1);
-    CHECK(i3->compare(*i2) == 1);
+    REQUIRE(x->compare(*x) == 0);
```
Comment (ShikharJ) [inline symengine/tests/basic/test_basic.cpp]: So can the rest of the cases be changed to `CHECK`?

Comment (isuruf) [inline symengine/tests/basic/test_basic.cpp]: It should, but that's a big change and I would only change it for the future

File: symengine/tests/basic/test_basic.cpp [hunk 2]
```diff
@@ -845,6 +851,45 @@ TEST_CASE("Complex: Basic", "[basic]")
...
+
+    c1 = Complex::from_two_nums(*integer(2), *integer(5));
+    c2 = Rational::from_two_ints(5, 5);
+    c3 = Complex::from_two_nums(*integer(2), *integer(5));
+    s = c1->div(*c2);
+    REQUIRE(eq(*s, *c3));
+
+    c1 = Complex::from_two_nums(*integer(2), *integer(5));
+    c2 = real_double(0.8);
+    CHECK_THROWS_AS(c1->rsub(*c2), NotImplementedError);
```
Comment (ShikharJ) [inline symengine/tests/basic/test_basic.cpp]: I must have accidentally made that. I'll change it.

File: symengine/tests/basic/test_basic.cpp (hunk 1 above)
Comment (ShikharJ) [inline symengine/tests/basic/test_basic.cpp]: Ok. I'd like to take that up whenever it's relevant.

File: symengine/tests/cwrapper/test_cwrapper.c [hunk 3]
```diff
@@ -85,6 +85,14 @@ void test_cwrapper()
     SYMENGINE_C_ASSERT(is_a_Rational(e));
     SYMENGINE_C_ASSERT(!is_a_Integer(e));
 
+    rational_set_si(e, 100, 47);
+    s = basic_str(e);
```
Comment (isuruf) [inline symengine/tests/cwrapper/test_cwrapper.c]: Value of `s` needs to be freed before assigning.

File: symengine/tests/basic/test_rational.cpp [hunk 4]
```diff
@@ -24,16 +30,24 @@ TEST_CASE("Rational", "[rational]")
...
+    REQUIRE(not q10_25->is_complex());
+
+    REQUIRE(is_a<Integer>(*r1));
+    REQUIRE(eq(*r1, *i2));
 
     r1 = rational(9, 3);
-    CHECK(is_a<Integer>(*r1));
-    CHECK(r1->__eq__(*integer(3)));
+    REQUIRE(is_a<Integer>(*r1));
+    REQUIRE(r1->__eq__(*integer(3)));
```
Comment (isuruf) [inline symengine/tests/basic/test_rational.cpp]: Can you change this back to the original?

**Comment by ShikharJ** [comment]
Stay tuned. More is to come :)

**Comment by isuruf** [comment]
@ShikarhJ, can you fix the merge conflicts?

**Comment by ShikharJ** [comment]
Updated @isuruf

**Pull request merged**